drivers/sensors/apds9922.c

```diff
@@ -0,0 +1,2598 @@
+ *
+ * Input Parameters:
+ *  priv - pointer to device structure
+ *  rate - LED current
+ *
+ * Returned Value:
+ *   Success or failure
+ *
+ ****************************************************************************/
+
+static int apds9922_ps_ledi(FAR struct apds9922_dev_s *priv,
+                            enum apds9922_ps_led_i current)
+{
+  uint8_t regval;
+  int ret;
+
+  if (current > PS_LED_CURRENT125MA)
+    {
+      return -EINVAL;
+    }
+
+  ret = apds9922_i2c_read8(priv, APDS9922_PS_LED, &regval);
+  if (ret < 0)
+    {
+      return ret;
+    }
+
+  regval &= ~PS_LED_CURRENT_MASK;
+  regval |= PS_SET_LED_CURRENT(current);
+  ret = apds9922_i2c_write8(priv, APDS9922_PS_LED, regval);
+  if (ret < 0)
+    {
+      return ret;
+    }
+
+  priv->ps_setup.led_i = current;
+
+  return OK;
+}
+
+/****************************************************************************
+ * Name: apds9922_ps_ledpk
+ *
+ * Description:
+ *   Turn LED peaking on/off.
+ *
+ * Input Parameters:
+ *  priv   - pointer to device structure
+ *  enable - enable or disable peaking
+ *
+ * Returned Value:
+ *   Success or failure
+ *
+ ****************************************************************************/
+
+static int apds9922_ps_ledpk(FAR struct apds9922_dev_s *priv, bool enable)
+{
+  uint8_t regval;
+  int ret;
+
+  ret = apds9922_i2c_read8(priv, APDS9922_PS_LED, &regval);
+  if (ret < 0)
+    {
+      return ret;
+    }
+
+  if (enable)
+    {
+      regval |= PS_LED_PEAKING_ON;
+    }
+  else
+    {
+      regval &= ~PS_LED_PEAKING_ON;
+    }
+
+  ret = apds9922_i2c_write8(priv, APDS9922_PS_LED, regval);
+  if (ret < 0)
+    {
+      return ret;
+    }
+
+  priv->ps_setup.led_pk_on = enable;
+
+  return OK;
+}
+
+/****************************************************************************
+ * Name: apds9922_ps_pulses
+ *
+ * Description:
+ *   Set the number of LED pulses.
+ *
+ * Input Parameters:
+ *  priv   - pointer to device structure
+ *  num_p  - the number of pulses
+ *
+ * Returned Value:
+ *   Success or failure
+ *
+ ****************************************************************************/
+
+static int apds9922_ps_pulses(FAR struct apds9922_dev_s *priv, uint8_t num_p)
+{
+  int ret;
+
+  ret = apds9922_i2c_write8(priv, APDS9922_PS_PULSES, num_p);
+
+  if (ret < 0)
+    {
+      return ret;
+    }
+
+  priv->ps_setup.pulses = num_p;
+
+  return OK;
+}
+
+/****************************************************************************
+ * Name: apds9922_ps_thresh
+ *
+ * Description:
+ *   Sets the PS thresholds, upper and lower.
+ *
+ * Input Parameters:
+ *  priv       - pointer to device structure
+ *  thresholds - struct of thresholds to set
+ *
+ * Returned Value:
+ *   Success or failure
+ *
+ ****************************************************************************/
+
+static int apds9922_ps_thresh(FAR struct apds9922_dev_s *priv,
+                              FAR struct adps9922_ps_thresh thresholds)
```

Comment:
```suggestion
                              struct adps9922_ps_thresh thresholds)
```
Since `thresholds` is not a pointer